edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,114 @@
+mod child_process;
+
+use std::{
+    io::Error,
+    process::{exit, Command, Stdio},
+    sync::atomic::{AtomicBool, Ordering},
+    sync::Arc,
+    thread::{sleep, spawn, JoinHandle},
+    time::Duration,
+};
+
+use child_process::ChildProcess;
+use child_process::PROCESS_POLL_INTERVAL_SECS;
+use signal_hook::{flag::register, SIGINT, SIGTERM};
+use tracing::{error, info, subscriber, Level};
+use tracing_subscriber::fmt::Subscriber;
+
+fn main() -> Result<(), Error> {
+    init_logging();
+    info!("Starting watchdog");
+
+    let should_shutdown = match register_shutdown_listener() {
+        Ok(should_shutdown) => should_shutdown,
+        Err(e) => {
+            error!(
+                "Failed to register sigterm listener. Shutting down. {:?}",
+                e
+            );
+            exit(1);
+        }
+    };
+
+    let broker_handle = run_child_process(
+        "MQTT Broker".to_string(),
+        "/usr/local/bin/mqttd".to_string(),
+        "-c /tmp/mqtt/config/production.json".to_string(),
+        Arc::clone(&should_shutdown),
+    );
+
+    let edgehub_handle = run_child_process(
+        "Edge Hub".to_string(),
+        "dotnet".to_string(),
+        "/app/Microsoft.Azure.Devices.Edge.Hub.Service.dll".to_string(),
+        Arc::clone(&should_shutdown),
+    );
+
+    match broker_handle.join() {
+        Ok(()) => info!("Successfully stopped broker process"),
```

Comment:
let's move this log message to the end of `run` function so the code will be a bit simpler
```rust
if let Err(e) = broker_handle.join() {
  should_shutdown.store(true, Ordering::Relaxed);
  error!("Failure while running broker process. {}", e)
}
```